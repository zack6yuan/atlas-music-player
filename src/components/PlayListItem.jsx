export default function PlayListItem() {
    return (
        <div className="playlist-items">
            <p>Song Title (Playlist)</p>
            <p>Song Artist (Playlist)</p>
        </div>
    )
}