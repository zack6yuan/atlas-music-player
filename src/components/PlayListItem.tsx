import { useState, useEffect, useRef } from "react";
import SongTitle from "./SongTitle";

export default function PlayListItem({
  changeFirstText,
  changeSecondText,
  changeThirdText,
  changeFourthText,
  changeFifthText,
  changeSixthText,
  changeSeventhText,
  changeEighthText,
  changeNinthText,
  changeTenthText,
  changeFirstArtist
}) {
  const [color, setColor] = useState("white");

  const handleText = () => {
    changeFirstText();
  }

  const playlistURL = "api/v1/playlist"; // playlist URL

  const songs = [
    {
      title: "Painted in Blue",
      artist: "Soul Canvas",
      length: "05:55",
    },
    {
      title: "Tidal Drift",
      artist: "Echoes of the Sea",
      length: "08:02",
    },
    {
      title: "Facing Shadows",
      artist: "The Emberlight",
      length: "03:01",
    },
    {
      title: "Cosmic Drift",
      artist: "Solar Flare",
      length: "05:01",
    },
    {
      title: "Urban Seranade",
      artist: "Midnight Groove",
      length: "04:54",
    },
    {
      title: "Whispers in the Wind",
      artist: "Rust & Ruin",
      length: "06:13",
    },
    {
      title: "Electric Fever",
      artist: "Neon Jungle",
      length: "08:41",
    },
    {
      title: "Electric Wildflower",
      artist: "Velvet Ember",
      length: "02:27",
    },
    {
      title: "Golden Haze",
      artist: "Velvet Waves",
      length: "03:15",
    },
    {
      title: "Shatter the Silence",
      artist: "Thunderclap Echo",
      length: "08:22",
    },
  ];

  return (
    <div className="">
      <h1 className="bg-white sm:mt-6 sm:ml-5">Playlist</h1>
      <div className="sm:-mt-6 md:-mt-8">
        <div
          id="0"
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          onClick={ changeFirstText }
        >
          <div className="font-medium sm:mt-8 2xl:mt-13">
            <p className="desc-text uppercase">{songs[0].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[0].artist}
            </p>
          </div>
          <div className="sm:mt-11 sm:mr-1 sm:ml-auto xl:mt-16 xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[0].length}
            </p>
          </div>
        </div>
        <div
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeSecondText}
        >
          <div className="font-medium">
            <p className="desc-text uppercase">{songs[1].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[1].artist}
            </p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[1].length}
            </p>
          </div>
        </div>
        <div
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeThirdText}
        >
          <div className="font-medium">
            <p className="desc-text uppercase">{songs[2].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[2].artist}
            </p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[2].length}
            </p>
          </div>
        </div>
        <div
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeFourthText}
        >
          <div className="font-medium">
            <p className="desc-text text-base uppercase">{songs[3].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[3].artist}
            </p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[3].length}
            </p>
          </div>
        </div>
        <div
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeFifthText}
        >
          <div className="font-medium">
            <p className="desc-text text-base uppercase">{songs[4].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[4].artist}
            </p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[4].length}
            </p>
          </div>
        </div>
        <div
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeSixthText}
        >
          <div className="font-medium">
            <p className="desc-text text-base uppercase">{songs[5].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[5].artist}
            </p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[5].length}
            </p>
          </div>
        </div>
        <div
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeSeventhText}
        >
          <div className="font-medium">
            <p className="desc-text text-base uppercase">{songs[6].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[6].artist}
            </p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[6].length}
            </p>
          </div>
        </div>
        <div
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeEighthText}
        >
          <div className="font-medium">
            <p className="desc-text text-base uppercase">{songs[7].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[7].artist}
            </p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[7].length}
            </p>
          </div>
        </div>
        <div
          className="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeNinthText}
        >
          <div className="font-medium">
            <p className="desc-text text-base uppercase">{songs[8].title}</p>
            <p className="text-crimson desc-text italic">{songs[8].artist}</p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[8].length}
            </p>
          </div>
        </div>
        <div
          className="rousong-div nded flex cursor-pointer sm:mt-1 sm:mr-9 sm:mb-5 sm:ml-5"
          style={{ backgroundColor: color }}
          onClick={changeTenthText}
        >
          <div className="font-medium">
            <p className="desc-text text-base uppercase">{songs[9].title}</p>
            <p className="text-crimson desc-text text-sm italic">
              {songs[9].artist}o
            </p>
          </div>
          <div className="sm:mt-3 sm:mr-1 sm:ml-auto xl:mr-40 2xl:mr-110">
            <p className="text-gray song-length font-medium">
              {songs[9].length}
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}
