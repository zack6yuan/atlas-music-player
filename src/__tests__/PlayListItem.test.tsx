import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import PlayListItem from "../components/PlayListItem"

test("Play List Item renders correctly", () => {
    const { container } = render(
        <PlayListItem />
    );
    expect(container).toMatchSnapshot();
});