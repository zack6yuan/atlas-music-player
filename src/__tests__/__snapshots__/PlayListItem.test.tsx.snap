// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Play List Item renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <h1
      class="bg-white sm:mt-6 sm:ml-5"
    >
      Playlist
    </h1>
    <div
      class="sm:-mt-6 md:-mt-8 lg:-mt-5 xl:-mt-10"
    >
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        id="0"
      >
        <div
          class="font-medium sm:mt-8 xl:mt-14 2xl:mt-13"
        >
          <p
            class="desc-text uppercase"
          >
            Painted in Blue
          </p>
          <div
            class=""
          >
            <p
              class="text-crimson desc-text text-sm italic"
            >
              Soul Canvas
            </p>
          </div>
        </div>
        <div
          class="sm:mt-11 sm:mr-1 sm:ml-auto xl:mt-16 xl:mr-2 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            05:55
          </p>
        </div>
      </div>
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text uppercase"
          >
            Tidal Drift
          </p>
          <div
            class=""
          >
            <p
              class="text-crimson desc-text text-sm italic"
            >
              Echoes of the Sea
            </p>
          </div>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            08:02
          </p>
        </div>
      </div>
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text uppercase"
          >
            Facing Shadows
          </p>
          <p
            class="text-crimson desc-text text-sm italic"
          >
            The Emberlight
          </p>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            03:01
          </p>
        </div>
      </div>
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text text-base uppercase"
          >
            Cosmic Drift
          </p>
          <p
            class="text-crimson desc-text text-sm italic"
          >
            Solar Flare
          </p>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            05:01
          </p>
        </div>
      </div>
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text text-base uppercase"
          >
            Urban Seranade
          </p>
          <p
            class="text-crimson desc-text text-sm italic"
          >
            Midnight Groove
          </p>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            04:54
          </p>
        </div>
      </div>
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text text-base uppercase"
          >
            Whispers in the Wind
          </p>
          <p
            class="text-crimson desc-text text-sm italic"
          >
            Rust & Ruin
          </p>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            06:13
          </p>
        </div>
      </div>
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text text-base uppercase"
          >
            Electric Fever
          </p>
          <p
            class="text-crimson desc-text text-sm italic"
          >
            Neon Jungle
          </p>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            08:41
          </p>
        </div>
      </div>
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text text-base uppercase"
          >
            Electric Wildflower
          </p>
          <p
            class="text-crimson desc-text text-sm italic"
          >
            Velvet Ember
          </p>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            02:27
          </p>
        </div>
      </div>
      <div
        class="song-div flex cursor-pointer rounded sm:mt-1 sm:mr-9 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text text-base uppercase"
          >
            Golden Haze
          </p>
          <p
            class="text-crimson desc-text italic"
          >
            Velvet Waves
          </p>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            03:15
          </p>
        </div>
      </div>
      <div
        class="rousong-div nded flex cursor-pointer sm:mt-1 sm:mr-9 sm:mb-5 sm:ml-5"
        style="background-color: white;"
      >
        <div
          class="font-medium"
        >
          <p
            class="desc-text text-base uppercase"
          >
            Shatter the Silence
          </p>
          <p
            class="text-crimson desc-text text-sm italic"
          >
            Thunderclap Echo
            o
          </p>
        </div>
        <div
          class="xl:mr2 sm:mt-3 sm:mr-1 sm:ml-auto 2xl:mr-110"
        >
          <p
            class="text-gray song-length font-medium"
          >
            08:22
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
